src/decomp_2d_init_fin.f90

```diff
@@ -232,34 +211,33 @@ end subroutine decomp_2d_init_ref
   subroutine decomp_2d_finalize_ref
 
      implicit none
-#if defined(_GPU) && defined(_NCCL)
-     type(ncclResult) :: nccl_stat
-#endif
+!#if defined(_GPU) && defined(_NCCL)
+!     type(ncclResult) :: nccl_stat
+!#endif
 
 #ifdef PROFILER
      if (decomp_profiler_d2d) call decomp_profiler_start("decomp_2d_fin")
 #endif
 
-     call decomp_mpi_comm_free(DECOMP_2D_COMM_ROW)
-     call decomp_mpi_comm_free(DECOMP_2D_COMM_COL)
-     call decomp_mpi_comm_free(DECOMP_2D_COMM_CART_X)
-     call decomp_mpi_comm_free(DECOMP_2D_COMM_CART_Y)
-     call decomp_mpi_comm_free(DECOMP_2D_COMM_CART_Z)
+     call decomp_2d_mpi_comm_free(DECOMP_2D_COMM_ROW)
+     call decomp_2d_mpi_comm_free(DECOMP_2D_COMM_COL)
+     call decomp_2d_mpi_comm_free(DECOMP_2D_COMM_CART_X)
+     call decomp_2d_mpi_comm_free(DECOMP_2D_COMM_CART_Y)
+     call decomp_2d_mpi_comm_free(DECOMP_2D_COMM_CART_Z)
 
      call decomp_info_finalize(decomp_main)
 
      decomp_buf_size = 0
      deallocate (work1_r, work2_r, work1_c, work2_c)
 #if defined(_GPU)
-     deallocate (work1_r_d, work2_r_d, work1_c_d, work2_c_d)
-
+     call decomp_2d_cumpi_fin()
 #if defined(_NCCL)
-     nccl_stat = ncclCommDestroy(nccl_comm_2decomp)
+     call decomp_2d_nccl_fin()
+     !nccl_stat = ncclCommDestroy(nccl_comm_2decomp)
```

Comment:
Delete?

Reply:
We can change the name, not really an issue for me. Don’t remember why i chose destroy over another name

Reply:
I meant the commented out line, not a comment on the name :)